staging/src/k8s.io/legacy-cloud-providers/gce/gce_loadbalancer_internal.go

```diff
@@ -980,3 +996,30 @@ func forwardingRulesEqual(old, new *compute.ForwardingRule) bool {
 		old.AllowGlobalAccess == new.AllowGlobalAccess &&
 		old.Subnetwork == new.Subnetwork
 }
+
+// patchILBNameAnnotation patches the annotations on the service with the provided ilbName.
+// If ilbName is empty or protocol is nil, the forwarding annotation keys will be deleted from
+// resource's annotation map
+func patchILBNameAnnotation(client clientset.Interface, service *v1.Service, protocol *v1.Protocol, ilbName string) (*v1.Service, error) {
```

Comment:
while we are at it, should we also add the annotations for backend-service, firewall rules and healthcheck?

Reply:
That is a good idea. Added logic for the backend service, firewall rules and healthcheck annotations